resolvers/userResolver.ts

```diff
@@ -51,60 +52,62 @@ export class UserResolver {
   }
 
   @Query(returns => User, { nullable: true })
-  userByAddress (@Arg('address', type => String) address: string) {
+  userByAddress(@Arg('address', type => String) address: string) {
     return this.userRepository.findOne({ walletAddress: address })
   }
 
   @Mutation(returns => Boolean)
-  async updateUser (
+  async updateUser(
     @Arg('firstName', { nullable: true }) firstName: string,
     @Arg('lastName', { nullable: true }) lastName: string,
     @Arg('location', { nullable: true }) location: string,
     @Arg('email', { nullable: true }) email: string,
-    @Arg('name', { nullable: true }) name: string,
+    @Arg('name') name: string,
     @Arg('url', { nullable: true }) url: string,
     @Ctx() { req: { user } }: MyContext
   ): Promise<boolean> {
     if (!user) throw new Error('Authentication required.')
     const dbUser = await User.findOne({ id: user.userId })
+    if (!dbUser) {
+      return false
+    }
+    if (!firstName && !lastName) {
```

Comment:
Is it fine if it's one or the other? Just curious about that.

Reply:
Yeah exaclty you can check this link to see it's part of our DOD

https://github.com/Giveth/giveth-next/issues/348#issuecomment-955265371